numba/core/codegen.py

```diff
@@ -535,12 +536,18 @@ def __init__(self, codegen: "BaseCPUCodegen", name: str):
         self._shared_module = None
         # Track names of the dynamic globals
         self._dynamic_globals = []
+        ptc_name = f"{self.__class__.__name__}({self._name!r})"
+        self._recorded_timings = PassTimingsCollection(ptc_name)
```

Comment:
Is there a performance impact for always collecting results?

Reply:
The maximum slowdown I measured is a 1.2x. The bigger the function, the less significant the slowdown is. I have a big function with a 1.06x slowdown. All measured with ipython timeit.